---
title: "Class 11: Genome Informatics"
format: pdf
editor: visual
---

```{r}
knitr::opts_chunk$set(echo = TRUE)
```

# Section 2 Proportion of G/G in a population


# Downloaded CSV file from Ensemble < 

Here we read this CSV file

```{r}
mxl <- read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378 (1).csv")
head(mxl)
```

```{r}
table(mxl$Genotype..forward.strand.)
```

```{r}
table(mxl$Genotype..forward.strand.) / nrow(mxl) * 100
```

## Section 4: Population Scale Analysis
One sample is obviously not enough to know what is happening in a population. You are
interested in assessing genetic differences on a population scale. So, you processed about
~230 samples and did the normalization on a genome level. Now, you want to find whether
there is any association of the 4 asthma-associated SNPs (rs8067378...) on ORMDL3
expression.

```{r}
expr <- read.table("https://bioboot.github.io/bimm143_S23/class-material/rs8067378_ENSG00000172057.6.txt")
head(expr)
```

```{r}
nrow(expr)
```

> Q13: Read this file into R and determine the sample size for each genotype and their corresponding median expression levels for each of these genotypes.

```{r}
table(expr$geno)

```
```{r}
expr$geno == "A/A"
median(expr$exp)
```



```{r}
library(ggplot2)

```

Let's make a boxplot

```{r}
ggplot(expr) + 
  aes(x=geno, y=exp, fill=geno) + 
  geom_boxplot(notch = TRUE)
```


